### PR TITLE
Fix parquet reader when fileOffset == 0

### PR DESCRIPTION
**Pull request by zhejiangxiaomai**
When we pass spark UT called "SPARK-34167".  The data of the case is a parquet file. It has 10 rows of Decimal(9, 1).  And rowGroups_[0].columns[0].file_offset is 0.  `VELOX_CHECK_GT(fileOffset, 0);` will throw exception.

file_offset of zero is expected.

### CODE REVIEW

**Comment by zhejiangxiaomai** [comment]
Hi, @Yuhta  would you take a look.  a very small fix.

File: velox/dwio/parquet/reader/ParquetReader.cpp [hunk 1]
```diff
@@ -562,7 +562,11 @@ void ParquetRowReader::filterRowGroups() {
     auto fileOffset = rowGroups_[i].__isset.file_offset
         ? rowGroups_[i].file_offset
         : rowGroups_[i].columns[0].file_offset;
-    VELOX_CHECK_GT(fileOffset, 0);
+    VELOX_CHECK_GE(fileOffset, 0);
+    if (fileOffset == 0) {
```
Comment (Yuhta) [inline velox/dwio/parquet/reader/ParquetReader.cpp:566]: We still need to pass the metadata filters in this case, do not shortcut this one

Comment (zhejiangxiaomai) [reply velox/dwio/parquet/reader/ParquetReader.cpp]: Thanks, just fixed.

**Review by bikramSingh91 — APPROVED** [review]
@zhejiangxiaomai Thank you for working on this fix. Can you please clarify in the PR description whether a file_offset of zero is expected and whether the check VELOX_CHECK_GT(fileOffset, 0) was actually a bug.

**Comment by zhejiangxiaomai** [comment]
> @zhejiangxiaomai Thank you for working on this fix. Can you please clarify in the PR description whether a file_offset of zero is expected and whether the check VELOX_CHECK_GT(fileOffset, 0) was actually a bug.

Updated.

**Pull request closed without merging**